### DISCUSSION TITLE
Filtering based on relation

### DISCUSSION THREAD

**Discussion by Sand-NGR** [pocketbase/pocketbase#3776] [Q&A] [answered]
I have a `posts` table and one of the columns in it, is a relation (multiple) to another table called `categories` like this:

- table 1 ( `posts`) --> one of the columns called `post_categories` which is a multiple relation to table 2.
- table 2 (`categories`) --> one of the columns called `category` which is just a plain text field with some data like "javascript".

I can get a post from pocketbase but how can I filter to get the posts that have the same category as the current post I've gotten?

this gives me an "Invalid filter parameters." error:
```ts
    const posts = await pb.collection('posts').getList<BlogPostData>(1, 2, {
      cache: 'no-store',
      filter: pb.filter('post_categories ?= categories.category'),
    });
```

This gives me no data back:
```ts
    const posts = await pb.collection('posts').getList<BlogPostData>(1, 2, {
      cache: 'no-store',
      filter: pb.filter('post_categories ?= {:category}', {
        category: 'javascript',
      }),
    });
```

**Comment by ganigeorgiev** [comment, marked as answer]
The first request returns "Invalid filter parameters" because the targeted collection "posts" doesn't have `categories.*` field (_aka. it tries to look for `categories` field in the "posts" schema_).

The second query doesn't return data because the `post_categories` field is the raw relation value (in your case serialized json array of ids). If you want to search by a relation subfield you have to append the specific subfield with dot-notation, something like:

```js
const posts = await pb.collection('posts').getList<BlogPostData>(1, 2, {
  cache: 'no-store',
  filter: pb.filter('post_categories.category ?= {:category}', {
    category: 'javascript',
  }),
});
```

**Reply by Sand-NGR** [reply]
Ah! Thank you.
